lib/errors.js

```diff
@@ -24,8 +24,8 @@ module.exports = class RPCError extends Error {
     return new RPCError(msg, 'CHANNEL_DESTROYED', RPCError.CHANNEL_DESTROYED)
   }
 
-  static REQUEST_ERROR (msg) {
-    return new RPCError(msg, 'REQUEST_ERROR', RPCError.REQUEST_ERROR)
+  static REQUEST_ERROR (msg, code) {
+    return new RPCError(msg, code, RPCError.REQUEST_ERROR)
```

Comment:
```suggestion
  static REQUEST_ERROR (msg, cause) {
    return new RPCError(msg,'REQUEST_ERROR', RPCError.REQUEST_ERROR, { cause })
```

The constructor should forward the options, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause.